tests/functional/testplan/testing/multitest/test_suite_decorators.py

```diff
@@ -7,22 +7,33 @@
 from testplan.common.utils.logger import TESTPLAN_LOGGER
 from testplan.report.testing import (TestReport, TestGroupReport,
                                      TestCaseReport)
+from testplan.testing.multitest.base import Categories
 from testplan.testing.multitest.suite import (testcase, testsuite, skip_if,
                                               post_testcase, pre_testcase)
 
 
-def pre(name, self, env, result):
+def pre1(name, self, env, result):
     result.equal(2, 2)
     result.contain('case', name)
 
 
-def post(name, self, env, result):
+def post1(name, self, env, result):
     result.equal(2, 2)
     result.contain('case', name)
 
 
-@pre_testcase(pre)
-@post_testcase(post)
+def pre2(name, self, env, result, a=None, b=None):
```

Comment:
Is my understanding correct, that `a` and `b` will be set when the pre/post testcase functions are run before the parametrized testcase, but will be None for other non-parametrized functions in the suite?

Reply:
Yes, that's right. If the suite only has parametrized testcases based on "case4", then we don't need that default value for 'a' and 'b', but the suite also has a normal "case5".

Reply:
Makes sense, thanks.